src/utils/WebSocketsService.ts

```diff
@@ -46,7 +37,9 @@ function initWebSocketConnection() {
 async function send(message) {
   // Lazily initialize connection
   await initWebSocketConnection();
-  socket.send(JSON.stringify(message));
+  if (socket && socket.readyState === 1) {
+    socket.send(JSON.stringify(message));
+  }
```

Comment:
I don't think we should do this because it hides failures. If the socket isn't ready, initWebSocketConnection should throw. We should check it in onopen and reject the promise in that case? Either that or throw here.